spec/models/order_detail_spec.rb

```diff
@@ -967,6 +967,16 @@ def assign_price_policy
         end
       end
     end
+
+    describe "non_cancelled" do
+      let!(:order_detail) { create(:order_detail, order_status: OrderStatus.canceled.first, order: Order.last, product: Product.last) }
```

Comment:
I wonder if it it would be worth adding `canceled` and `complete` traits onto the factory. Not necessary, but just a thought.